probe-rs/src/probe/list.rs

```diff
@@ -50,25 +53,27 @@ impl Default for Lister {
 /// Trait for a probe lister implementation.
 ///
 /// This trait can be used to implement custom probe listers.
-pub trait ProbeLister: std::fmt::Debug {
+#[async_trait::async_trait]
+pub trait ProbeLister: std::fmt::Debug + Send + Sync {
     /// Try to open a probe using the given selector
-    fn open(&self, selector: &DebugProbeSelector) -> Result<Probe, DebugProbeError>;
+    async fn open(&self, selector: &DebugProbeSelector) -> Result<Probe, DebugProbeError>;
 
     /// List all probes found by the lister.
-    fn list_all(&self) -> Vec<DebugProbeInfo> {
-        self.list(None)
+    async fn list_all(&self) -> Vec<DebugProbeInfo> {
+        self.list(None).await
     }
 
     /// List probes found by the lister, with optional filtering.
-    fn list(&self, selector: Option<&DebugProbeSelector>) -> Vec<DebugProbeInfo>;
+    async fn list(&self, selector: Option<&DebugProbeSelector>) -> Vec<DebugProbeInfo>;
 }
 
 /// Default lister implementation that includes all built-in probe drivers.
 #[derive(Debug, PartialEq, Eq)]
 pub struct AllProbesLister;
 
+#[async_trait::async_trait]
```

Comment:
Why do we need `async_trait`?

Reply:
Ah I see `The stabilization of async functions in traits in Rust 1.75 did not include support for using traits containing async functions as dyn Trait. Trying to use dyn with an async trait produces the following error:`

Reply:
We want to box the `Lister` trait which requires it to be dyn compatible, which means we cannot have generics on the trait functions. This makes it so that we need to return a boxed dyn Future which async_trait does for us, because without it the boilerplate is a bit of a hassle.